### DISCUSSION TITLE
Ninja Buddy .Config Controls

### DISCUSSION THREAD

**Discussion by TLable** [MicroTrendsLtd/NinjaTrader8#70] [Ideas]
I am attempting a config update, and wonder if it is possible to hard code a click into the config menu and then update a value in the properties to allow a seamless DateTime or TimeSpan value to be updated and continual calculate rather than require a update using Refresh or restart.
I have a filter to show trend on a larger period chart, that also is used as an entry condition if it is set that is in use and will only update a max number of bars on screen when starting. I am able to increment the TimeDate value to a new larger number but the visuals do not update as a restart and the visuals still hit a max at 200-500 bars calculated. 
So If I used the GUI.Config button on a chart top bar, I am able to manually update the TimeDate value which will allow the system visuals to be updated as it was a new instance of the system starting. So I like to know if I am able to call the GUI's Config button click, while doing so set the increment TimeDate value to then, just on a countdowntimer, automatically update visuals or every x number of bars calculated. 
Is there function ability to call user functions or update a value while the button's config method is called.

**Comment by MicroTrendsTom** [comment]
The system ships as is as open source, if you got the imagination and skills you can modify it in anyway you want - there are some technical limitations that will allow or disallow ideas- trial and error is the way to find out.
For general programming questions you can try a community on programming such as stackoverflow.com for WPF and C#

Also specific to NinjaTrader 8 - the online guidance and forum 
and in platform samples e.g. a bar timer might show a good pattern for a WPF dispatcher to avoid cross threading issues and locks etc

**Comment by TLable** [comment]
Help here is very much appreciated, can someone tell me if a cs file that is attched to how the NinjaBuddy.xaml is compiled and written that allows me to open a workspace in VS & customize the custom chart trader. Which may be easier than instead picking apart the xaml file, seems changes made to xaml negate the already implimented custom chart trader features.

**Comment by MicroTrendsTom** [comment]
The WPF project is for devs with basic level commercial WPF skills -  so that is for advanced NinjaTrader users only etc
As detailed prior on WPF project etc:

you can load and edit that WPF project 
1) deploy it overwrite the source 
or
2) compile it and add in a ref and remove the Ninjabuddy open source and then bind the events and so forth etc.

if you dont understand that part then you will need to learn WPF or hire a dev to do that part 
That is not something can that be taught here it would require a lot of time and effort to learn from scratch etc.
